README.md

```diff
@@ -1,84 +1,143 @@
-## blocklock-solidity
+# blocklock-solidity
 
-This repository contains the Solidity-based smart contracts library that facilitates Randamu's on-chain timelock encryption and decryption.
+[![Solidity ^0.8.x](https://img.shields.io/badge/Solidity-%5E0.8.x-blue)](https://soliditylang.org/)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green)](LICENSE)
+[![Foundry Tests](https://img.shields.io/badge/Tested%20with-Foundry-red)](https://book.getfoundry.sh/)
 
-By leveraging this library, developers can implement time-based data unlocking mechanisms securely in their smart contracts.
+A Solidity library enabling on-chain blocklock encryption and decryption, from the [dcipher threshold network](https://dcipher.network/). This facilites secure time-based data unlocking mechanisms for smart contracts.
 
-This library is designed with modularity and simplicity in mind, allowing developers to extend and integrate it into their existing projects easily.
+## ✨ Overview
+
+Controlling access to data based on time is crucial for various use cases, such as auctions, voting, and content release schedules. `blocklock-solidity` provides developers with tools to implement blocklock encryption on-chain, ensuring that encrypted data can only be decrypted after a specified block height, thus enhancing security and fairness in time-sensitive operations. 
+
+This blocklock library is powered by the dciper threshold network using **BLS pairing-based signature scheme** and **identity-based encryption** to achive data encryption toward the future block height without relaying on the trust of a thrid party.  It’s especially useful in decentralized settings where there's no trusted third party to enforce timing.
+
+The library is designed with modularity and simplicity in mind, allowing developers to easily integrate it into their existing smart contract projects to achieve blocklock on-chain. Its extensible architecture makes it suitable for a wide range of applications that require robust on-chain randomness.
 
 ### Features
-* Timelock Encryption: Encrypt data that can only be decrypted after a specified block number.
-* Decryption Callback: Implement custom logic that gets triggered when the decryption key is received, i.e., decryption of the Ciphertext.
-* Abstract Interface: Extend and implement the library to suit your specific needs.
 
+Powered by the dcipher threshold network and its threshold-based cryptographic schemes, this library offers:
+* **On-chain Blocklock Encryption**: Encrypt data that can only be decrypted after a specified block number.
```

Comment:
Timelock Encryption


We will also support other condition types, other than blocknumbers.

Reply:
agree with Tari here - blocklock is a lib name for timelock encryption